transform-plugins/src/main/java/co/cask/hydrator/plugin/CSVParser.java

```diff
@@ -176,26 +176,26 @@ public void transform(StructuredRecord in, Emitter<StructuredRecord> emitter) th
       org.apache.commons.csv.CSVParser parser = org.apache.commons.csv.CSVParser.parse(body, csvFormat);
       List<CSVRecord> records = parser.getRecords();
       for (CSVRecord record : records) {
-        if (fields.size() == record.size()) {
-          StructuredRecord sRecord = createStructuredRecord(record);
-          emitter.emit(sRecord);
-        } else {
-          LOG.warn("Skipping record as output schema specified has '{}' fields, while CSV record has '{}'",
-                   fields.size(), record.size());
-          // Write the record to error Dataset.
-        }
+        emitter.emit(createStructuredRecord(record, in));
       }
     } catch (IOException e) {
-      LOG.error("There was a issue parsing the record. ", e.getLocalizedMessage());
+      throw new RuntimeException("There was a issue parsing the record. " + e.getLocalizedMessage());
     }
   }
 
-  private StructuredRecord createStructuredRecord(CSVRecord record) {
+  private StructuredRecord createStructuredRecord(CSVRecord record, StructuredRecord in) {
     StructuredRecord.Builder builder = StructuredRecord.builder(outSchema);
     int i = 0;
     for (Field field : fields) {
-      builder.set(field.getName(), TypeConvertor.get(record.get(i), field.getSchema().getType()));
-      ++i;
+      String name = field.getName();
+      // If the field specified in the output field is present in the input, then
+      // it's directly copied into the output, else field is parsed in from the CSV parser.
+      if (in.get(name) != null) {
+        builder.set(name, in.get(name));
```

Comment:
Will check for the type before adding it, but I think I will have to add something to check during ```Configure(...)```